components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -706,32 +706,33 @@ private String getCaptchaParams(String tenantDomain, int failedLoginAttempts) {
-                            Property maxFailedProperty = maxFailedConfig[0];
-                            try {
-                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Number of failed attempts is higher than max failed " +
-                                                "attempts for reCaptcha. Recaptcha will be enforced");
-                                    }
-                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
-                                } else {
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Number of failed attempts is less than max failed " +
-                                                "attempts for reCaptcha. Recaptcha will not be enforced");
-                                    }
+                        if (!Boolean.parseBoolean(connectorConfig.getValue())) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Enforcing recaptcha for SSO Login is not enabled.");
+                            }
+                            continue;
+                        }
+                        Property[] maxFailedConfig =
+                                BasicAuthenticatorDataHolder.getInstance().getIdentityGovernanceService()
+                                        .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName},
+                                                tenantDomain);
+                        Property maxFailedProperty = maxFailedConfig[0];
+                        try {
+                            if (Integer.valueOf(maxFailedProperty.getValue()) >= failedLoginAttempts) {
+                                if (log.isDebugEnabled()) {
+                                    log.debug("Number of failed attempts is less than max failed " +
+                                            "attempts for reCaptcha. Recaptcha will not be enforced");
                                 }
-                            } catch (NumberFormatException e) {
-                                log.error(String.format("Invalid Max failed attempts for reCaptcha: %s",
-                                        maxFailedProperty.getValue()), e);
+                                continue;
                             }
-                        } else {
                             if (log.isDebugEnabled()) {
-                                log.debug("Enforcing recaptcha for SSO Login is not enabled.");
+                                log.debug("Number of failed attempts is higher than max failed " +
+                                        "attempts for reCaptcha. Recaptcha will be enforced");
```

Comment:
The log is wrong right? It should be lesser than max attempts right?

Reply:
No this is the true condition when failed login attempts exceeds the max failed attempt.  So log says, reCaptcha will be enforced.